homeassistant/components/sensor/recorder.py

```diff
@@ -284,6 +286,23 @@ def warn_dip(hass: HomeAssistant, entity_id: str) -> None:
         )
 
 
+def warn_negative(hass: HomeAssistant, entity_id: str) -> None:
+    """Log a warning once if a sensor with state_class_total has a negative value."""
+    if WARN_NEGATIVE not in hass.data:
+        hass.data[WARN_NEGATIVE] = set()
+    if entity_id not in hass.data[WARN_NEGATIVE]:
+        hass.data[WARN_NEGATIVE].add(entity_id)
+        domain = entity_sources(hass).get(entity_id, {}).get("domain")
+        _LOGGER.warning(
+            "Entity %s %shas state class total_increasing, but its state is "
+            "negative. Please create a bug report at %s",
+            entity_id,
+            f"from integration {domain} " if domain else "",
+            "https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue"
+            "+label%3A%22integration%3A+recorder%22",
```

Comment:
Shouldn't domain be the one of the integration? And if we link this, we should also link somewhere else if it's a custom component

Reply:
Fixed in 7cec8e1